### PR TITLE
standardize `^^^ x / !x vvv` comments

### PR DESCRIPTION
**Pull request by fsb4000**
Sometimes we use `// ^^^ no workaround / workaround vvv` and sometimes `// ^^^ no workaround // workaround vvv`.
It looks like `/` is used more often. So I replaced all of `//` which I had found.
I used `//.+//` regexp and VS code.
Driven by: https://github.com/microsoft/STL/pull/3206#discussion_r1020755546

### CODE REVIEW

**Comment by SuperWig** [comment]
Should this PR also settle on double arrows or one on each side i.e.
````
^^^ _HAS_CXX20 ^^^/ vvv !_HAS_CXX20 vvv
vs
^^^ _HAS_CXX20 / !_HAS_CXX20 vvv

**Comment by fsb4000** [comment]
Sure, and what do you like more?

**Comment by strega-nil-ms** [comment]
I'll take a look at this once we discuss #351

File: stl/inc/bitset [hunk 1]
```diff
@@ -69,7 +69,7 @@ public:
     static _CONSTEXPR23 void _Validate(const size_t _Pos) noexcept { // verify that _Pos is within bounds
 #if _ITERATOR_DEBUG_LEVEL == 0
         (void) _Pos;
-#else // ^^^ _ITERATOR_DEBUG_LEVEL == 0 ^^^ // vvv _ITERATOR_DEBUG_LEVEL != 0 vvv
+#else // ^^^ _ITERATOR_DEBUG_LEVEL == 0 ^^^ / vvv _ITERATOR_DEBUG_LEVEL != 0 vvv
```
Comment (strega-nil-ms) [inline stl/inc/bitset:72]:
```suggestion
#else // ^^^ _ITERATOR_DEBUG_LEVEL == 0 / _ITERATOR_DEBUG_LEVEL != 0 vvv
```

File: stl/inc/deque [hunk 2]
```diff
@@ -407,7 +407,7 @@ public:
     void _Verify_offset(const difference_type _Off) const noexcept {
 #if _ITERATOR_DEBUG_LEVEL == 0
         (void) _Off;
-#else // ^^^ _ITERATOR_DEBUG_LEVEL == 0 ^^^ // vvv _ITERATOR_DEBUG_LEVEL != 0 vvv
+#else // ^^^ _ITERATOR_DEBUG_LEVEL == 0 ^^^ / vvv _ITERATOR_DEBUG_LEVEL != 0 vvv
```
Comment (strega-nil-ms) [inline stl/inc/deque:410]:
```suggestion
#else // ^^^ _ITERATOR_DEBUG_LEVEL == 0 / _ITERATOR_DEBUG_LEVEL != 0 vvv
```

File: stl/inc/execution [hunk 3]
```diff
@@ -33,7 +33,7 @@ _EXTERN_C
 using __std_TP_WORK              = void;
 using __std_TP_CALLBACK_INSTANCE = void;
 using __std_TP_CALLBACK_ENVIRON  = void;
-#else // ^^^ _M_CEE ^^^ // vvv !_M_CEE vvv
+#else // ^^^ _M_CEE ^^^ / vvv !_M_CEE vvv
```
Comment (strega-nil-ms) [inline stl/inc/execution:36]:
```suggestion
#else // ^^^ _M_CEE / !_M_CEE vvv
```

File: stl/inc/execution [hunk 4]
```diff
@@ -2811,7 +2811,7 @@ inline size_t _Get_stable_sort_tree_height(const size_t _Count, const size_t _Hw
     const size_t _Log_ideal_chunks     = _Floor_of_log_2(_Ideal_chunks);
 #ifdef _WIN64
     const size_t _Max_tree_height      = 62; // to avoid ptrdiff_t overflow
-#else // ^^^ _WIN64 ^^^ // vvv !_WIN64 vvv
+#else // ^^^ _WIN64 ^^^ / vvv !_WIN64 vvv
```
Comment (strega-nil-ms) [inline stl/inc/execution:2814]:
```suggestion
#else // ^^^ _WIN64 / !_WIN64 vvv
```

File: stl/inc/list [hunk 5]
```diff
@@ -369,7 +369,7 @@ public:
                 *_Pnext             = *_Pnextnext;
             }
         }
-#else // ^^^ _ITERATOR_DEBUG_LEVEL == 2 ^^^ // vvv _ITERATOR_DEBUG_LEVEL != 2 vvv
+#else // ^^^ _ITERATOR_DEBUG_LEVEL == 2 ^^^ / vvv _ITERATOR_DEBUG_LEVEL != 2 vvv
```
Comment (strega-nil-ms) [inline stl/inc/list:372]:
```suggestion
#else // ^^^ _ITERATOR_DEBUG_LEVEL == 2 / _ITERATOR_DEBUG_LEVEL != 2 vvv
```

File: stl/inc/new [hunk 6]
```diff
@@ -101,7 +101,7 @@ _NODISCARD_LAUNDER constexpr _Ty* launder(_Ty* _Ptr) noexcept {
 #if defined(_M_IX86) || defined(_M_X64) || defined(_M_ARM) || defined(_M_ARM64)
 _EXPORT_STD inline constexpr size_t hardware_constructive_interference_size = 64;
 _EXPORT_STD inline constexpr size_t hardware_destructive_interference_size  = 64;
-#else // ^^^ supported hardware ^^^ // vvv unsupported hardware vvv
+#else // ^^^ supported hardware ^^^ / vvv unsupported hardware vvv
```
Comment (strega-nil-ms) [inline stl/inc/new:104]:
```suggestion
#else // ^^^ supported hardware / unsupported hardware vvv
```

File: stl/inc/numeric [hunk 7]
```diff
@@ -30,7 +30,7 @@ _NODISCARD _CONSTEXPR20 _Ty accumulate(const _InIt _First, const _InIt _Last, _T
     for (; _UFirst != _ULast; ++_UFirst) {
 #if _HAS_CXX20
         _Val = _Reduce_op(_STD move(_Val), *_UFirst);
-#else // ^^^ _HAS_CXX20 ^^^ // vvv !_HAS_CXX20 vvv
+#else // ^^^ _HAS_CXX20 ^^^ / vvv !_HAS_CXX20 vvv
```
Comment (strega-nil-ms) [inline stl/inc/numeric:33]:
```suggestion
#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
```

File: stl/inc/numeric [hunk 8]
```diff
@@ -62,7 +62,7 @@ _Ty _Reduce_plus_arithmetic_ranges(_InIt _First, const _InIt _Last, _Ty _Val) {
 }
 #pragma float_control(pop)
 
-#else // ^^^ _STD_VECTORIZE_WITH_FLOAT_CONTROL ^^^ // vvv !_STD_VECTORIZE_WITH_FLOAT_CONTROL vvv
+#else // ^^^ _STD_VECTORIZE_WITH_FLOAT_CONTROL ^^^ / vvv !_STD_VECTORIZE_WITH_FLOAT_CONTROL vvv
```
Comment (strega-nil-ms) [inline stl/inc/numeric:65]:
```suggestion
#else // ^^^ _STD_VECTORIZE_WITH_FLOAT_CONTROL / !_STD_VECTORIZE_WITH_FLOAT_CONTROL vvv
```

File: stl/inc/numeric [hunk 9]
```diff
@@ -134,7 +134,7 @@ _NODISCARD _CONSTEXPR20 _Ty inner_product(
     for (; _UFirst1 != _ULast1; ++_UFirst1, (void) ++_UFirst2) {
 #if _HAS_CXX20
         _Val = _Reduce_op(_STD move(_Val), _Transform_op(*_UFirst1, *_UFirst2)); // Requirement missing from N4713
-#else // ^^^ _HAS_CXX20 ^^^ // vvv !_HAS_CXX20 vvv
+#else // ^^^ _HAS_CXX20 ^^^ / vvv !_HAS_CXX20 vvv
```
Comment (strega-nil-ms) [inline stl/inc/numeric:137]:
```suggestion
#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
```

File: stl/inc/numeric [hunk 10]
```diff
@@ -167,7 +167,7 @@ _Ty _Transform_reduce_arithmetic_defaults(_InIt1 _First1, const _InIt1 _Last1, _
     return _Val;
 }
 #pragma float_control(pop)
-#else // ^^^ _STD_VECTORIZE_WITH_FLOAT_CONTROL ^^^ // vvv !_STD_VECTORIZE_WITH_FLOAT_CONTROL vvv
+#else // ^^^ _STD_VECTORIZE_WITH_FLOAT_CONTROL ^^^ / vvv !_STD_VECTORIZE_WITH_FLOAT_CONTROL vvv
```
Comment (strega-nil-ms) [inline stl/inc/numeric:170]:
```suggestion
#else // ^^^ _STD_VECTORIZE_WITH_FLOAT_CONTROL / !_STD_VECTORIZE_WITH_FLOAT_CONTROL vvv
```

File: stl/inc/numeric [hunk 11]
```diff
@@ -259,7 +259,7 @@ _CONSTEXPR20 _OutIt partial_sum(const _InIt _First, const _InIt _Last, _OutIt _D
             }
 #if _HAS_CXX20
             _Val = _Reduce_op(_STD move(_Val), *_UFirst);
-#else // ^^^ _HAS_CXX20 ^^^ // vvv !_HAS_CXX20 vvv
+#else // ^^^ _HAS_CXX20 ^^^ / vvv !_HAS_CXX20 vvv
```
Comment (strega-nil-ms) [inline stl/inc/numeric:262]:
```suggestion
#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
```

File: stl/inc/numeric [hunk 12]
```diff
@@ -483,7 +483,7 @@ _CONSTEXPR20 _OutIt adjacent_difference(const _InIt _First, const _InIt _Last, _
             _Iter_value_t<_InIt> _Tmp = *_UFirst;
 #if _HAS_CXX20
             *++_UDest = _Func(_Tmp, _STD move(_Val));
-#else // ^^^ _HAS_CXX20 ^^^ // vvv !_HAS_CXX20 vvv
+#else // ^^^ _HAS_CXX20 ^^^ / vvv !_HAS_CXX20 vvv
```
Comment (strega-nil-ms) [inline stl/inc/numeric:486]:
```suggestion
#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
```

File: stl/inc/span [hunk 13]
```diff
@@ -606,7 +606,7 @@ public:
     _NODISCARD constexpr iterator begin() const noexcept {
 #if _ITERATOR_DEBUG_LEVEL >= 1
         return {_Mydata, _Mydata, _Mydata + _Mysize};
-#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ // vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
+#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ / vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
```
Comment (strega-nil-ms) [inline stl/inc/span:609]:
```suggestion
#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 / _ITERATOR_DEBUG_LEVEL == 0 vvv
```

File: stl/inc/span [hunk 14]
```diff
@@ -615,7 +615,7 @@ public:
         const auto _End = _Mydata + _Mysize;
 #if _ITERATOR_DEBUG_LEVEL >= 1
         return {_End, _Mydata, _End};
-#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ // vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
+#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ / vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
```
Comment (strega-nil-ms) [inline stl/inc/span:618]:
```suggestion
#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 / _ITERATOR_DEBUG_LEVEL == 0 vvv
```

File: stl/inc/string [hunk 15]
```diff
@@ -445,7 +445,7 @@ _NODISCARD _Elem* _UIntegral_to_buff(_Elem* _RNext, _UTy _UVal) {
 
 #ifdef _WIN64
     auto _UVal_trunc = _UVal;
-#else // ^^^ _WIN64 ^^^ // vvv !_WIN64 vvv
+#else // ^^^ _WIN64 ^^^ / vvv !_WIN64 vvv
```
Comment (strega-nil-ms) [inline stl/inc/string:448]:
```suggestion
#else // ^^^ _WIN64 / !_WIN64 vvv
```

File: stl/inc/vector [hunk 16]
```diff
@@ -101,7 +101,7 @@ public:
     _CONSTEXPR20 void _Verify_offset(const difference_type _Off) const noexcept {
 #if _ITERATOR_DEBUG_LEVEL == 0
         (void) _Off;
-#else // ^^^ _ITERATOR_DEBUG_LEVEL == 0 ^^^ // vvv _ITERATOR_DEBUG_LEVEL != 0 vvv
+#else // ^^^ _ITERATOR_DEBUG_LEVEL == 0 ^^^ / vvv _ITERATOR_DEBUG_LEVEL != 0 vvv
```
Comment (strega-nil-ms) [inline stl/inc/vector:104]:
```suggestion
#else // ^^^ _ITERATOR_DEBUG_LEVEL == 0 / _ITERATOR_DEBUG_LEVEL != 0 vvv
```

File: stl/inc/vector [hunk 17]
```diff
@@ -188,7 +188,7 @@ public:
         // test for compatible iterator pair
 #if _ITERATOR_DEBUG_LEVEL == 0
         (void) _Right;
-#else // ^^^ _ITERATOR_DEBUG_LEVEL == 0 ^^^ // vvv _ITERATOR_DEBUG_LEVEL != 0 vvv
+#else // ^^^ _ITERATOR_DEBUG_LEVEL == 0 ^^^ / vvv _ITERATOR_DEBUG_LEVEL != 0 vvv
```
Comment (strega-nil-ms) [inline stl/inc/vector:191]:
```suggestion
#else // ^^^ _ITERATOR_DEBUG_LEVEL == 0 / _ITERATOR_DEBUG_LEVEL != 0 vvv
```

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]
Might as well standardize the location of the `^^^` and `vvv` as well; also do that through the rest of the file (probably should've just done one suggestion and then said that... sorry...)

**Review by strega-nil-ms — APPROVED** [review]

**Review by StephanTLavavej — APPROVED** [review]
Thanks for making this more consistent and easier to read! The changes to `<xcharconv_ryu.h>` are fine (they're in additions to the upstream code and don't represent unnecessary divergence).

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks again for this consistency pass! :smile_cat: :tada: :100:

**Pull request merged**